src/logger/junit.ts

```diff
@@ -0,0 +1,84 @@
+import { LogEntry } from './../types';
+import xmlEscape = require('xml-escape');
+
+enum Level {
+    WARNING = 'warning',
+    ERROR = 'error',
+}
+
+interface ReportData {
+    failures: number;
+    errors: number;
+    tests: number;
+    rules: { [rule: string]: string[] };
+}
+
+const reportTemplate = (suite: string) => `<?xml version="1.0" encoding="utf-8"?>
+    <testsuites package="stricter">
+        ${suite}
+    </testsuites>`;
+
+const suiteTemplate = (
+    name: string,
+    failureCount: number,
+    errors: number,
+    tests: number,
+    testcases: string[],
+) => `<testsuite name="${xmlEscape(
+    name,
+)}" failures="${failureCount}" errors="${errors}" tests="${tests}">
+    ${testcases.join('\n')}
+</testsuite>`;
+
+const testcaseTemplate = (name: string, failures: string) => `<testcase name="${xmlEscape(name)}">
+    ${failures}
+</testcase>`;
+
+const testcaseFailure = (type: Level, message: string, detail: string = '') =>
+    `<failure type="${type}" message="${xmlEscape(message)}">${
+        detail !== '' ? `<![CDATA[${detail}]]>` : ''
```

Comment:
/nit Although rare, `detail` needs to escape `']]>'` sequence when inside `CDATA`.